src/components/manager/access/Subject.tsx

```diff
@@ -0,0 +1,93 @@
+import type { CSSProperties } from "react";
+import { Link } from "react-router-dom";
+
+import { Button, Popover, Typography } from "antd";
+import { useAuthState } from "bento-auth-js";
+
+import { useGroupsByID } from "@/modules/authz/hooks";
+import type { GrantSubject, StoredGroup } from "@/modules/authz/types";
+import { stringifyJSONRenderIfMultiKey } from "./utils";
+import { CloseOutlined } from "@ant-design/icons";
+
+type InnerSubjectProps = {
+    subject: GrantSubject;
+    boldLabel?: boolean;
+};
+
+const InnerSubject = ({ subject, boldLabel }: InnerSubjectProps) => {
+    const { idTokenContents: currentIDToken } = useAuthState();
+
+    const groupsByID: Record<number, StoredGroup> = useGroupsByID();
+
+    const renderAsBold = boldLabel ?? true;  // default to true
+    const labelStyle = { fontWeight: renderAsBold ? "bold" : "normal" };
+
+    /*
+    There are four possible configurations of a subject:
+     - { iss: "<issuer>", sub: "<subject>" } (1)
+     - { iss: "<issuer>", client: "<client ID>" } (2)
+     - { group: <group ID> } --> group itself contains either an expression or a list of (1) and/or (2)
+     - { everyone: true }
+     */
+
+    if ("iss" in subject) {
+        const { iss } = subject;
+        const isSub = "sub" in subject;
+        return (
+            <p style={{ margin: 0, lineHeight: "1.6em" }}>
+                <span style={labelStyle}>{isSub ? "Subject" : "Client"}:</span>{" "}
+                <Typography.Text code={true}>{isSub ? subject.sub : subject.client}</Typography.Text><br />
+                <span style={labelStyle}>Issuer:</span>{" "}
+                <Typography.Text code={true}>{iss}</Typography.Text><br />
+                {(isSub && subject.sub === currentIDToken?.sub && iss === currentIDToken?.iss)
+                    ? <em>(this is you)</em>
+                    : null}
+            </p>
+        );
+    } else if ("group" in subject) {
+        const { group } = subject;
+        const groupDef = groupsByID[subject.group];
+        return (
+            <>
+                <span style={labelStyle}>Group:</span>{" "}
+                <Link to={`/data/manager/access/groups#group-${group}`}>
+                    {groupDef
+                        ? (<>{groupDef.name} (ID: {group})</>)
+                        : (<>ID: {group}</>)}
+                </Link>
+            </>
+        );
+    } else if ("everyone" in subject) {
+        return (
+            <Popover content="Everyone, even anonymous users."><span style={labelStyle}>Everyone</span></Popover>
```

Comment:
What use cases would we want to cover with grants assigned to `everyone`?
Assuming `everyone` has a grant to delete projects, would it be possible for an anonymous user to delete a project?
I'm worried about anonymous users having dangerous permissions.

Reply:
maybe i should put a big warning? and maybe we can forbid some permissions from being everyone-enabled... but for example, RENATA wants all data to be publicly discoverable, and eventually we should have all discovery endpoints use the bool/counts permissions

Reply:
I've added a warning, but maybe it should also not be the default choice...